.github/workflows/publish-to-docker-hub.yml

```diff
@@ -0,0 +1,50 @@
+on:
+  workflow_dispatch:
+    environment: docker-publication
```

Comment:
What does using an environment here give us?

Reply:
well the TEST_SECRET one I need to delete but the COMMIT_HASH one just makes it availabe in the script for

`./docker-build.sh "$COMMIT_HASH"`

I think I could also just do:

`./docker-buid.sh {inputs.commit}`

I'm honestly not sure which is better but the tutorial I followed mapped the inputs and secrets to env for some reason, so really just copy/paste